user-impl/src/main/java/com/example/auction/user/impl/CredentialEntity.java

```diff
@@ -0,0 +1,45 @@
+package com.example.auction.user.impl;
+
+import com.lightbend.lagom.javadsl.persistence.PersistentEntity;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public class CredentialEntity extends PersistentEntity<CredentialCommand, CredentialEvent, Optional<PCredential>> {
```

Comment:
I think this is a situation where you would use a read-side processor. The credentials can be stored in the user entity, and a read-side processor can update a separate credential registry table with username and password fields. Then the login service call would query that table.

I like to create a class that encapsulates both the update and the query of the read-side table. In this case, you could create a class called `CredentialsRegistry` that registers an event processor in its constructor. This is the same pattern used in [`ItemRepository.java`](https://github.com/lagom/online-auction-java/blob/master/item-impl/src/main/java/com/example/auction/item/impl/ItemRepository.java).